app/models/user.rb

```diff
@@ -0,0 +1,40 @@
+require 'bcrypt'
+
+class User
+  include DataMapper::Resource
+  attr_reader :password
+  attr_accessor :password_confirmation
+
+  property :id, Serial
+  property :username, String, required: true
+  property :name, String, required: true
+  property :email, String, required: true
+  property :password_digest, Text
+
+  validates_presence_of :username
+  validates_presence_of :email
+  validates_presence_of :name
+  validates_confirmation_of :password
+  validates_uniqueness_of :email
+  validates_uniqueness_of :username
```

Comment:
Prefer the new style validations `validates :column, uniqueness: value` over `validates_uniqueness_of`.